engine/table/src/main/java/io/deephaven/stream/StreamToBlinkTableAdapter.java

```diff
@@ -277,50 +212,49 @@ public Table table() {
 
     @Override
     public void close() {
-        if (!alive) {
-            return;
-        }
-        synchronized (this) {
-            if (!alive) {
-                return;
-            }
-            alive = false;
+        if (alive.compareAndSet(true, false)) {
             log.info().append("Deregistering ").append(StreamToBlinkTableAdapter.class.getSimpleName()).append('-')
                     .append(name)
                     .endl();
             updateSourceRegistrar.removeSource(this);
-            final Runnable localShutdownCallback = shutdownCallback;
-            if (localShutdownCallback != null) {
-                localShutdownCallback.run();
-            }
+            streamPublisher.shutdown();
         }
     }
 
-    @Override
-    public void destroy() {
-        close();
-    }
-
     @Override
     public void run() {
+        synchronized (this) {
+            // If we have an enqueued failure we want to process it first, before we allow the streamPublisher to flush
+            // itself.
+            if (enqueuedFailures != null) {
+                deliverFailure(MultiException.maybeWrapInMultiException(
+                        "Multiple errors encountered while ingesting stream",
+                        enqueuedFailures.toArray(new Throwable[0])));
+                return;
+            }
+        }
         final TableUpdate downstream;
         try {
             downstream = doRefresh();
         } catch (Exception e) {
             log.error().append("Error refreshing ").append(StreamToBlinkTableAdapter.class.getSimpleName()).append('-')
                     .append(name).append(": ").append(e).endl();
-            updateSourceRegistrar.removeSource(this);
             deliverFailure(e);
             return;
         }
         deliverUpdate(downstream);
     }
 
-    private void deliverUpdate(final TableUpdate downstream) {
+    private void deliverUpdate(@Nullable final TableUpdate downstream) {
         final QueryTable localTable = tableRef.get();
         if (localTable != null) {
             if (downstream != null) {
-                localTable.notifyListeners(downstream);
+                try {
+                    localTable.notifyListeners(downstream);
+                } catch (Exception e) {
```

Comment:
I notice here, and other places in this impl, we are specifically concerning ourselves w/ catching `Exception`, even though the downstream code is more general to accept `Throwable`. Practically, this means that means that `OutOfMemoryError`s won't get caught - I wonder, is this intentional, or just how it's been done before? Should we consider catching Throwable?

Reply:
Typically, the idea is that `Error` is uncaught and allowed to kill the process.